source/adios2/ADIOSTypes.h

```diff
@@ -137,6 +114,35 @@ struct TypeInfo<T, typename std::enable_if<std::is_same<
     using ValueType = typename T::value_type;
 };
 
+const size_t UnknownDim = 0;
+const size_t JoinedDim = SIZE_MAX - 1;
```

Comment:
That was generally seen as an oversight and defect in c++98. c+11 made these all constexpr so they are evaluated at compile time.